### ISSUE TITLE
Compilation error in erlware_commons on Elixir v1.2.1

### ISSUE THREAD

**Issue by alco** [bitwalker/exrm#294]
When trying to build [master](https://github.com/bitwalker/exrm/commit/864d9bf00680d01e256e1dbfac09f87093d45d47) using Elixir v1.2.1, I get this:

```
==> erlware_commons
Error evaluating rebar config script ./rebar.config.script:16: evaluation failed with reason error:{badmatch,undefined} and stacktrace [{erl_eval,expr,3,[]}]
Any dependency defined in the script won't be available unless you add them to your Mix project
```

The fix is already in [master of erlware_commons](https://github.com/erlware/erlware_commons/commit/b84eed36342b932b0cd0b7f0dfca4e7fcd8f4d3a), but not yet published.

**Comment by parubets** [comment]
hey! faced similar problem, quick fix:

`{:cf, "~> 0.2.1", override: true}, # remove this later
 {:erlware_commons, github: "erlware/erlware_commons", override: true}, # remove this later`

**Comment by mrluc** [comment]
@parubets I tried that -- was depending on exrm master, and attempting your fix I get errors like:

```
WARN:  /Users/mrluc/projects/mine/elixir/convoke/_build/prod/lib/erlware_commons/ebin/erlware_commons.app has version "0.18.0"; requested regex was 0.15.0
Dependency not available: erlware_commons-0.15.0 (undefined)
ERROR: compile failed while processing /Users/mrluc/projects/mine/elixir/convoke/deps/relx: rebar_abort
** (Mix) Could not compile dependency :relx, "/usr/local/bin/rebar compile skip_deps=true deps_dir="/Users/mrluc/projects/mine/elixir/convoke/_build/prod/lib"" command failed. You can recompile this dependency with "mix deps.compile relx", update it with "mix deps.update relx" or clean it with "mix deps.clean relx"
```

**Comment by johnhamelink** [comment]
@parubets's temporary fix worked for me, FWIW. @mrluc did you try deleting your `_build` and `deps` directories and running `mix do deps.get, compile`

**Comment by bitwalker** [comment]
@alco @parubets @mrluc @johnhamelink A fix is coming to relx for this, I'll update exrm to use the new version as soon as it's released.

**Comment by mrluc** [comment]
Still getting the following:

```
mrluc@airy:someapp$ rm -rf deps/
mrluc@airy:someapp$ rm -rf _build/
mrluc@airy:someapp$ mix deps.get
... (successfully got) ...
mrluc@airy:someapp$ mix compile
... lots of output, then,
WARN:  .../_build/dev/lib/erlware_commons/ebin/erlware_commons.app has version "0.18.0"; requested regex was 0.15.0
WARN:  .../_build/dev/lib/erlware_commons/ebin/erlware_commons.app has version "0.18.0"; requested regex was 0.15.0
==> relx (compile)
WARN:  /Users/mrluc/projects/mine/elixir/convoke/_build/dev/lib/erlware_commons/ebin/erlware_commons.app has version "0.18.0"; requested regex was 0.15.0
WARN:  .../_build/dev/lib/erlware_commons/ebin/erlware_commons.app has version "0.18.0"; requested regex was 0.15.0
Dependency not available: erlware_commons-0.15.0 (undefined)
ERROR: compile failed while processing .../deps/relx: rebar_abort
** (Mix) Could not compile dependency :relx, "/usr/local/bin/rebar compile skip_deps=true deps_dir=".../_build/dev/lib"" command failed. You can recompile this dependency with "mix deps.compile relx", update it with "mix deps.update relx" or clean it with "mix deps.clean relx"
```
Will look at it later

**Comment by mrluc** [comment]
@johnhamelink yeah, I just forgot that I also had to `mix deps.update erlware_commons` as they're github deps. 

TL;DR: needed to add an override git dependency on relx, and then `mix deps.clean --all && mix deps.update erlware_commons relx`

Using tsloughter's pull request, above, it updated the error to:
``` erlware_commons.app has version "0.18.0"; requested regex was 0.19.0```
I forked erlware_commons and bumped its version to 0.19.0 to ensure that it was just the version number causing trouble, and that worked.

Since it's just version number causing problems, and since relx master already uses correct version, adding a git dependency for relx makes the following git dependencies all work together for me right now:

```
      {:exrm, git: "git://github.com/bitwalker/exrm.git"},
      {:relx, github: "erlware/relx", override: true}, # <-- depends on erlware_commons-0.18.0
      {:erlware_commons, github: "erlware/erlware_commons", override: true}, # <-- @0.18.0 right now
      {:cf, "~> 0.2.1", override: true},
```

Can't wait for updated packages though ... to make releases work in my umbrella app, each subapp also needs those deps! :)

**Comment by developerworks** [comment]
+1 

Mac OSX 10.11
Elixir 1.2.1
Erlang/OTP 18

**Comment by sashaafm** [comment]
+1

Ubuntu 14.04
Elixir 1.2.1
Erlang/OTP 18

**Comment by bitwalker** [comment]
The commit to relx master has been merged, so you no longer need to override `erlware_commons`, just `relx` for now. Once a new release of relx is pushed, I will update exrm.

**Comment by tsloughter** [comment]
I published a new relx to hex.

**Comment by bitwalker** [comment]
Thanks @tsloughter!

I've pushed a new release of exrm to hex. 1.0.0 finally - I'm getting sick of -rc releases, and it's already used in production at my office as well as a bunch of other places, so it doesn't make sense to call it an -rc release anymore. Anyway, you all should be good to go now.

**Comment by katgironpe** [comment]
I get this same issue when I tried to upgrade build to use Erlang 1.9.

**Comment by jimmycuadra** [comment]
Can this be reopened pending release of a new relx, or is there another issue I should subscribe to for updates? My team has an app that can't compile against Erlang 19, seemingly because of this issue.

**Comment by bitwalker** [comment]
I published a new release today with an updated relx, should be good to go!
